### PR TITLE
feat: complete ckb4ibc for enabling test cases

### PR DESCRIPTION
**Pull request by ImJeremyHe**
Closes: #XXX

## Description

______

### PR author checklist:
- [ ] Added changelog entry, using [`unclog`](https://github.com/informalsystems/unclog).
- [ ] Added tests: integration (for Forcerelay) or unit/mock tests (for modules).
- [ ] Linked to GitHub issue.
- [ ] Updated code comments and documentation (e.g., `docs/`).
- [ ] Tagged *one* reviewer who will be the one responsible for shepherding this PR.

### Reviewer checklist:

- [ ] Reviewed `Files changed` in the GitHub PR explorer.
- [ ] Manually tested (in case integration/unit/mock tests are absent).

### CODE REVIEW

File: crates/relayer/src/chain/ckb4ibc.rs [hunk 1]
```diff
@@ -314,12 +334,20 @@ impl Ckb4IbcChain {
...
         &self,
     ) -> Result<(Vec<IdentifiedConnectionEnd>, IbcConnections, CellInput), Error> {
+        println!("chain endpoint query connection and cache");
         let script = Script::new_builder()
-            .code_hash(self.get_converter().get_connection_code_hash())
-            .args(self.config.client_id.to_vec().pack())
+            .code_hash(get_script_hash(&self.config.connection_type_args))
+            .args(
+                ConnectionArgs {
+                    client_id: self.config.client_type_args.clone().into(),
```
Comment (ashuralyk) [inline crates/relayer/src/chain/ckb4ibc.rs]: why do not use `self.config.client_type_args` to pass client_id parameter?

File: crates/relayer/src/chain/ckb4ibc.rs [hunk 2]
```diff
@@ -409,26 +444,61 @@ impl ChainEndpoint for Ckb4IbcChain {
...
             &config.client_type_args.as_bytes().to_owned(),
         ))?;
         if client_cell.is_none() {
             return Err(Error::other_error(
-                "invalid `lightclient_contract_typeargs` option".to_owned(),
+                "invalid `client type args not found` option".to_owned(),
+            ));
+        }
+
+        let conn_contract_cell = rt.block_on(rpc_client.search_cell_by_typescript(
```
Comment (ashuralyk) [inline crates/relayer/src/chain/ckb4ibc.rs:463]: just wrap them into utility tool, including the below process of channel and packet

File: crates/relayer/src/chain/ckb4ibc.rs [hunk 3]
```diff
@@ -489,34 +559,87 @@ impl ChainEndpoint for Ckb4IbcChain {
...
-            let (tx, envelope, input_capacity) = convert_msg_to_ckb_tx(msg, &converter)?;
-            if let Ok(tx) =
-                self.complete_tx_with_secp256k1_change_and_envelope(tx, input_capacity, envelope)
-            {
+            let CkbTxInfo {
+                unsigned_tx,
+                envelope,
+                input_capacity,
+                event,
+            } = convert_msg_to_ckb_tx(msg, &converter)?;
```
Comment (ashuralyk) [inline crates/relayer/src/chain/ckb4ibc.rs:575]: this function is called in a loop before sending to ckb network, so the assembled multi transactions can share a same sender input cell, please make sure this won't happen

File: crates/relayer/src/chain/ckb4ibc/message/conn.rs [hunk 4]
```diff
@@ -55,20 +71,37 @@ pub fn convert_conn_open_init_to_tx<C: MsgToTxConverter>(
...
-
     let packed_tx = TransactionView::new_advanced_builder()
+        .cell_dep(
+            CellDep::new_builder()
+                .dep_type(DepType::Code.into())
+                .out_point(converter.get_client_outpoint())
+                .build(),
+        )
+        .cell_dep(
+            CellDep::new_builder()
```
Comment (ashuralyk) [inline crates/relayer/src/chain/ckb4ibc/message/conn.rs:81]: you can just use converter to generate `CellDep` directly

File: tools/ckb4ibc-test/src/rpc_client.rs
Comment (ashuralyk) [inline tools/ckb4ibc-test/src/rpc_client.rs:1]: this file cannot use the exsited one?

File: tools/ckb4ibc-test/txs/create_connection.json
Comment (ashuralyk) [inline tools/ckb4ibc-test/txs/create_connection.json:1]: this is your native generated test file?

**Review by ashuralyk — COMMENTED** [review]
great work on this, just leaves packet feature to work on

File: tools/ckb4ibc-test/src/rpc_client.rs
Comment (ImJeremyHe) [reply tools/ckb4ibc-test/src/rpc_client.rs:1]: the existed one is only used in #[cfg(not(test))]

File: tools/ckb4ibc-test/txs/create_connection.json
Comment (ImJeremyHe) [reply tools/ckb4ibc-test/txs/create_connection.json:1]: As the REAMDE.md said, it is generated by a script from another repo.

**Review by ashuralyk — APPROVED** [review]

**Pull request merged**